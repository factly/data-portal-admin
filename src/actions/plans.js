import axios from '../utils/axios';
import {
  baseUrl,
  ADD_PLANS_LIST_REQUEST,
  SET_PLANS_LIST_CURRENT_PAGE,
  LOADING_PLANS,
  LOAD_PLANS_SUCCESS,
  SET_PLANS_LIST_TOTAL,
  LOAD_PLANS_FAILURE,
  GET_PLAN_SUCCESS,
  GET_PLAN_FAILURE,
  CREATING_PLAN,
  CREATE_PLAN_SUCCESS,
  CREATE_PLAN_FAILURE,
  UPDATING_PLAN,
  UPDATE_PLAN_SUCCESS,
  UPDATE_PLAN_FAILURE,
  DELETING_PLAN,
  DELETE_PLAN_SUCCESS,
  DELETE_PLAN_FAILURE,
} from '../constants/plans';
import { getIds, buildObjectOfItems } from '../utils/objects';

export const loadPlans = (page, limit) => {
  return async (dispatch, getState) => {
    let url = baseUrl;
    if (page && limit) {
      url = `${url}?page=${page}&limit=${limit}`;
    }

    const {
      plans: { req },
    } = getState();

    let found = false;
    let ids;
    for (let item of req) {
      if (item.page === page && item.limit === limit) {
        ids = [...item.ids];
        found = true;
      }
    }

    if (found) {
      dispatch(setListCurrentPage(ids));
      return;
    }

    dispatch(loadingPlans());

    const response = await axios({
      url: url,
      method: 'get',
    }).catch((error) => {
      dispatch(loadPlansFailure(error.message));
    });

    if (response) {
      const { nodes, total } = response.data;
      const currentPageIds = getIds(nodes);
      const req = { page: page, limit: limit, ids: currentPageIds };
      dispatch(addListRequest(req));
      dispatch(loadPlansSuccess(nodes));
      dispatch(setListCurrentPage(currentPageIds));
      dispatch(setPlansListTotal(total));
    }
  };
};

export const createPlan = (data) => {
  return async (dispatch, getState) => {
    dispatch(creatingPlan());

    const response = await axios({
      url: baseUrl,
      method: 'post',
      data: data,
    }).catch((error) => {
      dispatch(createPlanFailure(error.message));
    });

    if (response) {
      dispatch(createPlanSuccess(response.data));
    }
  };
};

export const updatePlan = (id, data) => {
  return async (dispatch, getState) => {
    let url = `${baseUrl}/${id}`;

    dispatch(updatingPlan());

    const response = await axios({
      url: url,
      method: 'put',
      data: data,
    }).catch((error) => {
      dispatch(updatePlanFailure(error.message));
    });

    if (response) {
      dispatch(updatePlanSuccess(response.data));
    }
  };
};

export const deletePlan = (id) => {
  return async (dispatch, getState) => {
    let url = `${baseUrl}/${id}`;

    dispatch(deletingPlan());

    const response = await axios({
      url: url,
      method: 'delete',
    }).catch((error) => {
      dispatch(deletePlanFailure(error.message));
    });

    if (response) {
      dispatch(deletePlanSuccess(id));
    }
  };
};

export const getPlan = (id) => {
  return async (dispatch, getState) => {
    const {
      plans: { items },
    } = getState();

    if (items[id]) {
      dispatch(getPlanSuccess({ ...items[id] }));
      return;
    }

    dispatch(loadingPlans());

    const response = await axios({
      url: `${baseUrl}/${id}`,
      method: 'get',
    }).catch((error) => {
      dispatch(getPlanFailure(error.message));
    });

    if (response) {
      dispatch(getPlanSuccess(response.data));
    }
  };
};

const getPlanSuccess = (plan) => {
  return {
    type: GET_PLAN_SUCCESS,
    payload: plan,
  };
};

const getPlanFailure = (message) => {
  return {
    type: GET_PLAN_FAILURE,
    payload: message,
  };
};

const loadingPlans = () => {
  return {
    type: LOADING_PLANS,
  };
};

const setPlansListTotal = (total) => {
  return {
    type: SET_PLANS_LIST_TOTAL,
    payload: total,
  };
};

const addListRequest = (req) => {
  return {
    type: ADD_PLANS_LIST_REQUEST,
    payload: req,
  };
};

const setListCurrentPage = (ids) => {
  return {
    type: SET_PLANS_LIST_CURRENT_PAGE,
    payload: ids,
  };
};

export const loadPlansSuccess = (plans) => {
  return {
    type: LOAD_PLANS_SUCCESS,
    payload: {
      items: buildObjectOfItems(plans),
    },
  };
};

const loadPlansFailure = (message) => {
  return {
    type: LOAD_PLANS_FAILURE,
    payload: message,
  };
};

const creatingPlan = () => {
  return {
    type: CREATING_PLAN,
  };
};

const createPlanSuccess = (plan) => {
  return {
    type: CREATE_PLAN_SUCCESS,
    payload: plan,
  };
};

const createPlanFailure = (message) => {
  return {
    type: CREATE_PLAN_FAILURE,
    payload: message,
  };
};

const updatingPlan = () => {
  return {
    type: UPDATING_PLAN,
  };
};

const updatePlanSuccess = (plan) => {
  return {
    type: UPDATE_PLAN_SUCCESS,
    payload: plan,
  };
};

const updatePlanFailure = (message) => {
  return {
    type: UPDATE_PLAN_FAILURE,
    payload: message,
  };
};

const deletingPlan = () => {
  return {
    type: DELETING_PLAN,
  };
};

const deletePlanSuccess = (id) => {
  return {
    type: DELETE_PLAN_SUCCESS,
    payload: id,
  };
};

const deletePlanFailure = (message) => {
  return {
    type: DELETE_PLAN_FAILURE,
    payload: message,
  };
};
