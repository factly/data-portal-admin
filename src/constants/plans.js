export const LOADING_PLANS = 'LOADING_PLANS';
export const LOAD_PLANS_SUCCESS = 'LOAD_PLANS_SUCCESS';
export const SET_PLANS_LIST_TOTAL = 'SET_PLANS_LIST_TOTAL';
export const LOAD_PLANS_FAILURE = 'LOAD_PLANS_FAILURE';
export const CREATING_PLAN = 'CREATING_PLAN';
export const CREATE_PLAN_SUCCESS = 'CREATE_PLAN_SUCCESS';
export const CREATE_PLAN_FAILURE = 'CREATE_PLAN_FAILURE';
export const UPDATING_PLAN = 'UPDATING_PLAN';
export const UPDATE_PLAN_SUCCESS = 'UPDATE_PLAN_SUCCESS';
export const UPDATE_PLAN_FAILURE = 'UPDATE_PLAN_FAILURE';
export const DELETING_PLAN = 'DELETING_PLAN';
export const DELETE_PLAN_SUCCESS = 'DELETE_PLAN_SUCCESS';
export const DELETE_PLAN_FAILURE = 'DELETE_PLAN_FAILURE';

export const baseUrl = `${process.env.REACT_APP_API_URL}/plans`;
