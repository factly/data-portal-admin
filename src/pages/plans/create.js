import React from 'react';
import { useDispatch } from 'react-redux';
import { useHistory } from 'react-router-dom';

import PlanCreateForm from './components/PlanCreateForm';
import { createPlan } from '../../actions/plans';

const PlanCreate = () => {
  const history = useHistory();
  const dispatch = useDispatch();

  const onCreate = async (values) => {
    await dispatch(createPlan(values));
    history.push('/plans');
  };

  return <PlanCreateForm onSubmit={onCreate} />;
};

export default PlanCreate;
