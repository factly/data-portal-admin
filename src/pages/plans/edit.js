import React from 'react';
import { useParams, useHistory } from 'react-router-dom';
import { useDispatch, useSelector } from 'react-redux';
import { Skeleton } from 'antd';

import PlanCreateForm from './components/PlanCreateForm';
import { getPlan, updatePlan } from '../../actions/plans';

const EditPlan = () => {
  const { id } = useParams();
  const history = useHistory();
  const dispatch = useDispatch();
  const { loading, plan } = useSelector(({ plans }) => ({
    loading: plans.loading,
    plan: plans.items[id],
  }));

  React.useEffect(() => {
    dispatch(getPlan(id));
  }, []);

  if (loading) return <Skeleton />;

  const onUpdate = async (values) => {
    await dispatch(updatePlan(id, values));
    history.push('/plans');
  };

  return <PlanCreateForm data={plan} onSubmit={onUpdate} />;
};

export default EditPlan;
