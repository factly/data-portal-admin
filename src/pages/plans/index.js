import React, { useState } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { Table, Input, Button, Popconfirm, Form, notification } from 'antd';
import moment from 'moment';
import { Link } from 'react-router-dom';
import { EditOutlined, DeleteOutlined, SaveOutlined, CloseOutlined } from '@ant-design/icons';

import { loadPlans, updatePlan, deletePlan } from '../../actions/plans';

const EditableCell = ({ editing, dataIndex, title, record, index, children, ...restProps }) => {
  return (
    <td {...restProps}>
      {editing ? (
        <Form.Item
          name={dataIndex}
          style={{
            margin: 0,
          }}
          rules={[
            {
              required: true,
              message: `Please enter ${title}!`,
            },
          ]}
        >
          <Input />
        </Form.Item>
      ) : (
        children
      )}
    </td>
  );
};

const Plans = (props) => {
  const [form] = Form.useForm();
  const [editingKey, setEditingKey] = useState('');
  const { data, total, load, update, remove } = props;

  React.useEffect(() => {
    load();
  }, [load]);

  const get = (page, limit) => {
    cancel();
    load(page, limit);
  };

  const isEditing = (record) => record.id === editingKey;

  const edit = (record) => {
    form.setFieldsValue(record);
    setEditingKey(record.id);
  };

  const cancel = () => {
    setEditingKey('');
  };

  const save = async (key) => {
    try {
      const row = await form.validateFields();
      const index = data.findIndex((item) => item.id === key);

      if (index > -1) {
        update(key, row, index)
          .then(() => {
            setEditingKey('');
            notification.success({
              message: 'Success',
              description: 'Plan succesfully updated',
            });
          })
          .catch(() => {
            notification.error({
              message: 'Error',
              description: 'Something went wrong',
            });
          });
      }
    } catch (err) {
      notification.warning({
        message: 'Warning',
        description: 'Validation failed',
      });
    }
  };

  const deletePlan = (key) => {
    const index = data.findIndex((item) => item.id === key);
    if (index > -1) {
      remove(key, index)
        .then(() => {
          notification.success({
            message: 'Success',
            description: 'Membership succesfully deleted',
          });
        })
        .catch(() => {
          notification.error({
            message: 'Error',
            description: 'Something went wrong',
          });
        });
    }
  };

  const columns = [
    {
      title: 'Name',
      dataIndex: 'plan_name',
      width: '20%',
      editable: true,
    },
    {
      title: 'Info',
      dataIndex: 'plan_info',
      width: '20%',
      editable: true,
    },
    {
      title: 'Status',
      dataIndex: 'status',
      width: '20%',
      editable: true,
    },
    {
      title: 'Created At',
      dataIndex: 'created_at',
      width: '20%',
      render: (_, record) => {
        return <span title={record.created_at}>{moment(record.created_at).fromNow()}</span>;
      },
    },
    {
      title: 'Operation',
      dataIndex: 'operation',
      render: (_, record) => {
        const editable = isEditing(record);
        return editable ? (
          <span>
            <Button
              type="primary"
              icon={<SaveOutlined />}
              onClick={() => save(record.id)}
              style={{
                marginRight: 8,
              }}
            >
              Save
            </Button>
            <Popconfirm title="Sure to cancel?" onConfirm={cancel}>
              <Button icon={<CloseOutlined />}>Cancel</Button>
            </Popconfirm>
          </span>
        ) : (
          <span>
            <Button
              type="primary"
              icon={<EditOutlined />}
              disabled={editingKey !== ''}
              onClick={() => edit(record)}
              style={{
                marginRight: 8,
              }}
            >
              Edit
            </Button>
            <Popconfirm
              disabled={editingKey !== ''}
              title="Sure to delete?"
              onConfirm={() => deletePlan(record.id)}
            >
              <Button disabled={editingKey !== ''} icon={<DeleteOutlined />}>
                Delete
              </Button>
            </Popconfirm>
          </span>
        );
      },
    },
  ];

  const mergedColumns = columns.map((col) => {
    if (!col.editable) {
      return col;
    }

    return {
      ...col,
      onCell: (record) => ({
        record,
        dataIndex: col.dataIndex,
        title: col.title,
        editing: isEditing(record),
      }),
    };
  });

  return (
    <div>
      <Link to={'/plans/create'}>
        <Button type="primary" style={{ marginBottom: 16 }}>
          Add Plan
        </Button>
      </Link>
      <Form form={form} component={false}>
        <Table
          components={{
            body: {
              cell: EditableCell,
            },
          }}
          bordered
          rowKey="id"
          dataSource={data}
          columns={mergedColumns}
          rowClassName="editable-row"
          pagination={{
            defaultPageSize: 5,
            onChange: get,
            total: total,
          }}
        />
      </Form>
    </div>
  );
};

Plans.propTypes = {
  data: PropTypes.array.isRequired,
  total: PropTypes.number.isRequired,
  load: PropTypes.func.isRequired,
  update: PropTypes.func.isRequired,
  remove: PropTypes.func.isRequired,
};

const mapStateToProps = (state) => {
  const { list } = state.plans;
  return {
    data: list.items,
    total: list.total,
  };
};

const mapDispatchToProps = (dispatch) => ({
  load: (page, limit) => dispatch(loadPlans(page, limit)),
  update: (id, data, index) => dispatch(updatePlan(id, data, index)),
  remove: (id, index) => dispatch(deletePlan(id, index)),
});

export default connect(mapStateToProps, mapDispatchToProps)(Plans);
