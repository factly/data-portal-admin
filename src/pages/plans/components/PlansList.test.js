import React from 'react';
import { BrowserRouter as Router, Link, useHistory } from 'react-router-dom';
import renderer, { act } from 'react-test-renderer';
import { useSelector, useDispatch } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { shallow, mount } from 'enzyme';
import { Popconfirm, Button, Table } from 'antd';

import '../../../matchMedia.mock';
import PlanList from './PlansList';
import { loadPlans, deletePlan } from '../../../actions/plans';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

jest.mock('react-redux', () => ({
  useSelector: jest.fn(),
  useDispatch: jest.fn(),
}));
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useHistory: jest.fn(),
}));
jest.mock('../../../actions/plans', () => ({
  loadPlans: jest.fn(),
  deletePlan: jest.fn(),
}));

describe('Plans List component', () => {
  let store;
  let mockedDispatch;
  const plan = {
    id: 1,
    plan_name: 'plan_name',
    plan_info: 'plan_info',
    status: 'status',
    created_at: '2020-12-12',
  };

  describe('snapshot testing', () => {
    beforeEach(() => {
      store = mockStore({});
      store.dispatch = jest.fn();
      mockedDispatch = jest.fn();
      useDispatch.mockReturnValue(mockedDispatch);
    });
    it('should render the component', () => {
      useSelector.mockImplementation((state) => ({}));
      const tree = renderer.create(<PlanList />).toJSON();
      expect(tree).toMatchSnapshot();
      expect(useSelector).toHaveBeenCalled();
    });
    it('should match component when loading', () => {
      useSelector.mockImplementation((state) => ({
        data: [],
        total: 0,
      }));
      const tree = renderer.create(<PlanList />).toJSON();
      expect(tree).toMatchSnapshot();
      expect(useSelector).toHaveBeenCalled();
    });
    it('should match component with plans', () => {
      useSelector.mockImplementation((state) => ({
        data: [plan],
        total: 1,
      }));

      let component;
      act(() => {
        component = renderer.create(
          <Router>
            <PlanList />
          </Router>,
        );
      });
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();

      expect(useSelector).toHaveBeenCalled();
      expect(mockedDispatch).toHaveBeenCalledTimes(1);
      expect(useSelector).toHaveReturnedWith({
        data: [plan],
        total: 1,
      });
      expect(loadPlans).toHaveBeenCalledWith(1, 5);
    });
  });
  describe('component testing', () => {
    beforeEach(() => {
      jest.clearAllMocks();
      mockedDispatch = jest.fn(() => new Promise((resolve) => resolve(true)));
      useDispatch.mockReturnValue(mockedDispatch);
    });
    it('should change the page', () => {
      useSelector.mockImplementation((state) => ({}));

      const wrapper = shallow(<PlanList />);
      const table = wrapper.find(Table);
      table.props().pagination.onChange(2);
      wrapper.update();
      const updatedTable = wrapper.find(Table);
      expect(updatedTable.props().pagination.current).toEqual(2);
    });
    it('should edit plan', () => {
      const push = jest.fn();
      useHistory.mockReturnValueOnce({ push });

      useSelector.mockImplementation((state) => ({
        data: [plan],
        total: 1,
      }));

      const wrapper = mount(
        <Router>
          <PlanList />
        </Router>,
      );
      const button = wrapper.find(Button).at(0);
      expect(button.text()).toEqual('Edit');
      button.simulate('click');
      expect(push).toHaveBeenCalledWith('/plans/1/edit');
    });
    it('should delete plan', () => {
      useSelector.mockImplementation((state) => ({
        data: [plan],
        total: 1,
      }));

      const wrapper = mount(
        <Router>
          <PlanList />
        </Router>,
      );
      const button = wrapper.find(Button).at(1);
      expect(button.text()).toEqual('Delete');

      button.simulate('click');
      const popconfirm = wrapper.find(Popconfirm);
      popconfirm
        .findWhere((item) => item.type() === 'button' && item.text() === 'OK')
        .simulate('click');
      expect(deletePlan).toHaveBeenCalled();
      expect(deletePlan).toHaveBeenCalledWith(1);
      expect(loadPlans).toHaveBeenCalledWith(1, 5);
    });
    it('should have no delete and edit buttons', () => {
      useSelector.mockImplementation((state) => ({}));

      const wrapper = mount(
        <Router>
          <PlanList />
        </Router>,
      );

      const button = wrapper.find(Button);
      expect(button.length).toEqual(0);
    });
  });
});
