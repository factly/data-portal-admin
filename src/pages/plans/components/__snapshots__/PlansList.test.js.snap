// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Plans List component snapshot testing should match component when loading 1`] = `
<div
  className="ant-table-wrapper"
>
  <div
    className="ant-spin-nested-loading"
  >
    <div
      className="ant-spin-container"
    >
      <div
        className="ant-table ant-table-bordered"
      >
        <div
          className="ant-table-container"
        >
          <div
            className="ant-table-content"
            onScroll={[Function]}
            style={Object {}}
          >
            <table
              style={
                Object {
                  "tableLayout": "auto",
                }
              }
            >
              <colgroup>
                <col
                  style={
                    Object {
                      "minWidth": "20%",
                      "width": "20%",
                    }
                  }
                />
                <col
                  style={
                    Object {
                      "minWidth": "20%",
                      "width": "20%",
                    }
                  }
                />
                <col
                  style={
                    Object {
                      "minWidth": "20%",
                      "width": "20%",
                    }
                  }
                />
                <col
                  style={
                    Object {
                      "minWidth": "20%",
                      "width": "20%",
                    }
                  }
                />
              </colgroup>
              <thead
                className="ant-table-thead"
              >
                <tr>
                  <th
                    className="ant-table-cell"
                    colSpan={null}
                    rowSpan={null}
                    style={Object {}}
                  >
                    Name
                  </th>
                  <th
                    className="ant-table-cell"
                    colSpan={null}
                    rowSpan={null}
                    style={Object {}}
                  >
                    Info
                  </th>
                  <th
                    className="ant-table-cell"
                    colSpan={null}
                    rowSpan={null}
                    style={Object {}}
                  >
                    Status
                  </th>
                  <th
                    className="ant-table-cell"
                    colSpan={null}
                    rowSpan={null}
                    style={Object {}}
                  >
                    Created At
                  </th>
                  <th
                    className="ant-table-cell"
                    colSpan={null}
                    rowSpan={null}
                    style={Object {}}
                  >
                    Operation
                  </th>
                </tr>
              </thead>
              <tbody
                className="ant-table-tbody"
              >
                <tr
                  className="ant-table-placeholder"
                  style={
                    Object {
                      "display": null,
                    }
                  }
                >
                  <td
                    className="ant-table-cell"
                    colSpan={5}
                    rowSpan={null}
                    style={Object {}}
                  >
                    <div
                      className="ant-empty ant-empty-normal"
                    >
                      <div
                        className="ant-empty-image"
                      >
                        <svg
                          className="ant-empty-img-simple"
                          height="41"
                          viewBox="0 0 64 41"
                          width="64"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <g
                            fill="none"
                            fillRule="evenodd"
                            transform="translate(0 1)"
                          >
                            <ellipse
                              className="ant-empty-img-simple-ellipse"
                              cx="32"
                              cy="33"
                              rx="32"
                              ry="7"
                            />
                            <g
                              className="ant-empty-img-simple-g"
                              fillRule="nonzero"
                            >
                              <path
                                d="M55 12.76L44.854 1.258C44.367.474 43.656 0 42.907 0H21.093c-.749 0-1.46.474-1.947 1.257L9 12.761V22h46v-9.24z"
                              />
                              <path
                                className="ant-empty-img-simple-path"
                                d="M41.613 15.931c0-1.605.994-2.93 2.227-2.931H55v18.137C55 33.26 53.68 35 52.05 35h-40.1C10.32 35 9 33.259 9 31.137V13h11.16c1.233 0 2.227 1.323 2.227 2.928v.022c0 1.605 1.005 2.901 2.237 2.901h14.752c1.232 0 2.237-1.308 2.237-2.913v-.007z"
                              />
                            </g>
                          </g>
                        </svg>
                      </div>
                      <p
                        className="ant-empty-description"
                      >
                        No Data
                      </p>
                    </div>
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Plans List component snapshot testing should match component with plans 1`] = `
<div
  className="ant-table-wrapper"
>
  <div
    className="ant-spin-nested-loading"
  >
    <div
      className="ant-spin-container"
    >
      <div
        className="ant-table ant-table-bordered"
      >
        <div
          className="ant-table-container"
        >
          <div
            className="ant-table-content"
            onScroll={[Function]}
            style={Object {}}
          >
            <table
              style={
                Object {
                  "tableLayout": "auto",
                }
              }
            >
              <colgroup>
                <col
                  style={
                    Object {
                      "minWidth": "20%",
                      "width": "20%",
                    }
                  }
                />
                <col
                  style={
                    Object {
                      "minWidth": "20%",
                      "width": "20%",
                    }
                  }
                />
                <col
                  style={
                    Object {
                      "minWidth": "20%",
                      "width": "20%",
                    }
                  }
                />
                <col
                  style={
                    Object {
                      "minWidth": "20%",
                      "width": "20%",
                    }
                  }
                />
              </colgroup>
              <thead
                className="ant-table-thead"
              >
                <tr>
                  <th
                    className="ant-table-cell"
                    colSpan={null}
                    rowSpan={null}
                    style={Object {}}
                  >
                    Name
                  </th>
                  <th
                    className="ant-table-cell"
                    colSpan={null}
                    rowSpan={null}
                    style={Object {}}
                  >
                    Info
                  </th>
                  <th
                    className="ant-table-cell"
                    colSpan={null}
                    rowSpan={null}
                    style={Object {}}
                  >
                    Status
                  </th>
                  <th
                    className="ant-table-cell"
                    colSpan={null}
                    rowSpan={null}
                    style={Object {}}
                  >
                    Created At
                  </th>
                  <th
                    className="ant-table-cell"
                    colSpan={null}
                    rowSpan={null}
                    style={Object {}}
                  >
                    Operation
                  </th>
                </tr>
              </thead>
              <tbody
                className="ant-table-tbody"
              >
                <tr
                  className="ant-table-row ant-table-row-level-0"
                  data-row-key={1}
                  onClick={[Function]}
                  style={Object {}}
                >
                  <td
                    className="ant-table-cell"
                    colSpan={null}
                    rowSpan={null}
                    style={Object {}}
                  >
                    plan_name
                  </td>
                  <td
                    className="ant-table-cell"
                    colSpan={null}
                    rowSpan={null}
                    style={Object {}}
                  >
                    plan_info
                  </td>
                  <td
                    className="ant-table-cell"
                    colSpan={null}
                    rowSpan={null}
                    style={Object {}}
                  >
                    status
                  </td>
                  <td
                    className="ant-table-cell"
                    colSpan={null}
                    rowSpan={null}
                    style={Object {}}
                  >
                    <span
                      title="2020-12-12"
                    >
                      in 3 months
                    </span>
                  </td>
                  <td
                    className="ant-table-cell"
                    colSpan={null}
                    rowSpan={null}
                    style={Object {}}
                  >
                    <span>
                      <button
                        className="ant-btn ant-btn-primary"
                        onClick={[Function]}
                        style={
                          Object {
                            "marginRight": 8,
                          }
                        }
                        type="button"
                      >
                        <span
                          aria-label="edit"
                          className="anticon anticon-edit"
                          role="img"
                        >
                          <svg
                            aria-hidden="true"
                            className=""
                            data-icon="edit"
                            fill="currentColor"
                            focusable="false"
                            height="1em"
                            viewBox="64 64 896 896"
                            width="1em"
                          >
                            <path
                              d="M257.7 752c2 0 4-.2 6-.5L431.9 722c2-.4 3.9-1.3 5.3-2.8l423.9-423.9a9.96 9.96 0 000-14.1L694.9 114.9c-1.9-1.9-4.4-2.9-7.1-2.9s-5.2 1-7.1 2.9L256.8 538.8c-1.5 1.5-2.4 3.3-2.8 5.3l-29.5 168.2a33.5 33.5 0 009.4 29.8c6.6 6.4 14.9 9.9 23.8 9.9zm67.4-174.4L687.8 215l73.3 73.3-362.7 362.6-88.9 15.7 15.6-89zM880 836H144c-17.7 0-32 14.3-32 32v36c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-36c0-17.7-14.3-32-32-32z"
                            />
                          </svg>
                        </span>
                        <span>
                          Edit
                        </span>
                      </button>
                      <button
                        className="ant-btn"
                        onClick={[Function]}
                        onMouseDown={[Function]}
                        onTouchStart={[Function]}
                        type="button"
                      >
                        <span
                          aria-label="delete"
                          className="anticon anticon-delete"
                          role="img"
                        >
                          <svg
                            aria-hidden="true"
                            className=""
                            data-icon="delete"
                            fill="currentColor"
                            focusable="false"
                            height="1em"
                            viewBox="64 64 896 896"
                            width="1em"
                          >
                            <path
                              d="M360 184h-8c4.4 0 8-3.6 8-8v8h304v-8c0 4.4 3.6 8 8 8h-8v72h72v-80c0-35.3-28.7-64-64-64H352c-35.3 0-64 28.7-64 64v80h72v-72zm504 72H160c-17.7 0-32 14.3-32 32v32c0 4.4 3.6 8 8 8h60.4l24.7 523c1.6 34.1 29.8 61 63.9 61h454c34.2 0 62.3-26.8 63.9-61l24.7-523H888c4.4 0 8-3.6 8-8v-32c0-17.7-14.3-32-32-32zM731.3 840H292.7l-24.2-512h487l-24.2 512z"
                            />
                          </svg>
                        </span>
                        <span>
                          Delete
                        </span>
                      </button>
                    </span>
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </div>
      <ul
        className="ant-pagination ant-table-pagination ant-table-pagination-right"
        style={Object {}}
        unselectable="unselectable"
      >
        <li
          aria-disabled={true}
          className="ant-pagination-prev ant-pagination-disabled"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex={null}
          title="Previous Page"
        >
          <a
            className="ant-pagination-item-link"
            disabled={true}
          >
            <span
              aria-label="left"
              className="anticon anticon-left"
              role="img"
            >
              <svg
                aria-hidden="true"
                className=""
                data-icon="left"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 000 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z"
                />
              </svg>
            </span>
          </a>
        </li>
        <li
          className="ant-pagination-item ant-pagination-item-1 ant-pagination-item-active"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex="0"
          title={1}
        >
          <a>
            1
          </a>
        </li>
        <li
          aria-disabled={true}
          className="ant-pagination-next ant-pagination-disabled"
          onClick={[Function]}
          onKeyPress={[Function]}
          tabIndex={null}
          title="Next Page"
        >
          <a
            className="ant-pagination-item-link"
            disabled={true}
          >
            <span
              aria-label="right"
              className="anticon anticon-right"
              role="img"
            >
              <svg
                aria-hidden="true"
                className=""
                data-icon="right"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
                />
              </svg>
            </span>
          </a>
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`Plans List component snapshot testing should render the component 1`] = `
<div
  className="ant-table-wrapper"
>
  <div
    className="ant-spin-nested-loading"
  >
    <div
      className="ant-spin-container"
    >
      <div
        className="ant-table ant-table-bordered"
      >
        <div
          className="ant-table-container"
        >
          <div
            className="ant-table-content"
            onScroll={[Function]}
            style={Object {}}
          >
            <table
              style={
                Object {
                  "tableLayout": "auto",
                }
              }
            >
              <colgroup>
                <col
                  style={
                    Object {
                      "minWidth": "20%",
                      "width": "20%",
                    }
                  }
                />
                <col
                  style={
                    Object {
                      "minWidth": "20%",
                      "width": "20%",
                    }
                  }
                />
                <col
                  style={
                    Object {
                      "minWidth": "20%",
                      "width": "20%",
                    }
                  }
                />
                <col
                  style={
                    Object {
                      "minWidth": "20%",
                      "width": "20%",
                    }
                  }
                />
              </colgroup>
              <thead
                className="ant-table-thead"
              >
                <tr>
                  <th
                    className="ant-table-cell"
                    colSpan={null}
                    rowSpan={null}
                    style={Object {}}
                  >
                    Name
                  </th>
                  <th
                    className="ant-table-cell"
                    colSpan={null}
                    rowSpan={null}
                    style={Object {}}
                  >
                    Info
                  </th>
                  <th
                    className="ant-table-cell"
                    colSpan={null}
                    rowSpan={null}
                    style={Object {}}
                  >
                    Status
                  </th>
                  <th
                    className="ant-table-cell"
                    colSpan={null}
                    rowSpan={null}
                    style={Object {}}
                  >
                    Created At
                  </th>
                  <th
                    className="ant-table-cell"
                    colSpan={null}
                    rowSpan={null}
                    style={Object {}}
                  >
                    Operation
                  </th>
                </tr>
              </thead>
              <tbody
                className="ant-table-tbody"
              >
                <tr
                  className="ant-table-placeholder"
                  style={
                    Object {
                      "display": null,
                    }
                  }
                >
                  <td
                    className="ant-table-cell"
                    colSpan={5}
                    rowSpan={null}
                    style={Object {}}
                  >
                    <div
                      className="ant-empty ant-empty-normal"
                    >
                      <div
                        className="ant-empty-image"
                      >
                        <svg
                          className="ant-empty-img-simple"
                          height="41"
                          viewBox="0 0 64 41"
                          width="64"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <g
                            fill="none"
                            fillRule="evenodd"
                            transform="translate(0 1)"
                          >
                            <ellipse
                              className="ant-empty-img-simple-ellipse"
                              cx="32"
                              cy="33"
                              rx="32"
                              ry="7"
                            />
                            <g
                              className="ant-empty-img-simple-g"
                              fillRule="nonzero"
                            >
                              <path
                                d="M55 12.76L44.854 1.258C44.367.474 43.656 0 42.907 0H21.093c-.749 0-1.46.474-1.947 1.257L9 12.761V22h46v-9.24z"
                              />
                              <path
                                className="ant-empty-img-simple-path"
                                d="M41.613 15.931c0-1.605.994-2.93 2.227-2.931H55v18.137C55 33.26 53.68 35 52.05 35h-40.1C10.32 35 9 33.259 9 31.137V13h11.16c1.233 0 2.227 1.323 2.227 2.928v.022c0 1.605 1.005 2.901 2.237 2.901h14.752c1.232 0 2.237-1.308 2.237-2.913v-.007z"
                              />
                            </g>
                          </g>
                        </svg>
                      </div>
                      <p
                        className="ant-empty-description"
                      >
                        No Data
                      </p>
                    </div>
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
